lib/mysql_framework/sql_query.rb

```diff
@@ -76,6 +76,19 @@ def bulk_values(bulk_values)
       self
     end
 
+    # This method is called to specify the values to bulk insert.
+    def bulk_upsert(columns)
+      @sql += 'ON DUPLICATE KEY UPDATE '
+
+      columns.each do |column|
+        @sql += "#{column} = values(#{column}), "
```

Comment:
I would prefer values to be capitalised as it is a reserved word and it would make it easier to read.